core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -0,0 +1,1018 @@
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+    let totalQuestions = questionPlayerEngineService.getQuestionCount();
+
+    expect(totalQuestions).toBe(3);
+  });
+
+  it('should return the exploration Id', () =>{
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+    let explorationId = questionPlayerEngineService.getExplorationId();
+
+    expect(explorationId).toBe('id1');
+  });
+
+  it('should return the version of exploration', () =>{
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+    let version = questionPlayerEngineService.getExplorationVersion();
+    expect(version).toBe(1);
+  });
+
+  it('should clear all questions', () =>{
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
```

Comment:
Just noting that this is a good test because it tests the "before" and "after".

Reply:
done